packages/app/components/Today/ProfessorCheckinButton.tsx

```diff
@@ -39,64 +37,81 @@ export default function TodayPageCheckinButton(): ReactElement {
           onCancel={() => setModalVisible(false)}
           okText="Check In"
           onOk={async () => {
+            const value = await form.validateFields();
             let redirectID: QueuePartial;
-            if (queueToCheckInto > -1) {
-              redirectID = await API.taStatus.checkIn(
-                Number(cid),
-                course?.queues[queueToCheckInto].room
-              );
-            } else {
-              redirectID = await API.taStatus.checkIn(
-                Number(cid),
-                queueToBeCreated
+            try {
+              if (queueToCheckInto > -1) {
+                redirectID = await API.taStatus.checkIn(
+                  Number(cid),
+                  course?.queues[queueToCheckInto].room
+                );
+              } else {
+                redirectID = await API.taStatus.checkIn(
+                  Number(cid),
+                  value.officeHourName
+                );
+              }
+              router.push(
+                "/course/[cid]/queue/[qid]",
+                `/course/${Number(cid)}/queue/${redirectID.id}`
               );
+            } catch (err) {
+              message.error(err.response?.data?.message);
             }
-            router.push(
-              "/course/[cid]/queue/[qid]",
-              `/course/${Number(cid)}/queue/${redirectID.id}`
-            );
           }}
         >
           <h3>Which queue would you like to check into?</h3>
           <Radio.Group
             value={queueToCheckInto}
             onChange={(e) => setQueueToCheckInto(e.target.value)}
           >
-            {course?.queues.map((q, i) => (
-              <ProfessorModalRadio key={q.id} value={i}>
-                {q.room}
-              </ProfessorModalRadio>
-            ))}
-            <ProfessorModalRadio value={-1}>
-              Other...
+            {course?.queues
```

Comment:
Thank you for adding validation, whoever worked on this code before is definitely grateful for that (perhaps I)